### PR TITLE
MINOR: Delegated the creation of new Thread to KafkaThread class.

### PR DESCRIPTION
**Pull request by kamalcph**

### CODE REVIEW

**Comment by kamalcph** [comment]
@ijuma 
Please take a look at this pull request.

File: clients/src/main/java/org/apache/kafka/common/utils/KafkaThread.java [hunk 1]
```diff
@@ -28,19 +28,19 @@
...
     }
 
     public KafkaThread(final String name, Runnable runnable, boolean daemon) {
         super(runnable, name);
-        configureThread(name, daemon);
+        configureThread(daemon);
     }
 
-    private void configureThread(final String name, boolean daemon) {
+    private void configureThread(boolean daemon) {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/utils/KafkaThread.java:39]: I think you still pass the name directly. Getting the name from the superclass involves a volatile read and it's unnecessary since we have it here.

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java [hunk 2]
```diff
@@ -444,14 +444,7 @@ public static String stackTrace(Throwable e) {
      * @return The unstarted thread
      */
     public static Thread newThread(String name, Runnable runnable, boolean daemon) {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/utils/Utils.java:446]: We should just remove this method.

File: log4j-appender/src/main/java/org/apache/kafka/log4jappender/KafkaLog4jAppender.java [hunk 3]
```diff
@@ -35,43 +33,43 @@
...
-    private static final String RETRIES_CONFIG = ProducerConfig.RETRIES_CONFIG;
-    private static final String KEY_SERIALIZER_CLASS_CONFIG = ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
-    private static final String VALUE_SERIALIZER_CLASS_CONFIG = ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
-    private static final String SECURITY_PROTOCOL = CommonClientConfigs.SECURITY_PROTOCOL_CONFIG;
-    private static final String SSL_TRUSTSTORE_LOCATION = SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG;
-    private static final String SSL_TRUSTSTORE_PASSWORD = SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG;
-    private static final String SSL_KEYSTORE_TYPE = SslConfigs.SSL_KEYSTORE_TYPE_CONFIG;
-    private static final String SSL_KEYSTORE_LOCATION = SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG;
-    private static final String SSL_KEYSTORE_PASSWORD = SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG;
-    private static final String SASL_KERBEROS_SERVICE_NAME = SaslConfigs.SASL_KERBEROS_SERVICE_NAME;
```
Comment (ijuma) [inline log4j-appender/src/main/java/org/apache/kafka/log4jappender/KafkaLog4jAppender.java:55]: This and `MockKafkaLog4jAppender` seems unrelated to the thread change.

**Review by ijuma — COMMENTED** [review]
Thanks for the PR, left some comments.

File: log4j-appender/src/main/java/org/apache/kafka/log4jappender/KafkaLog4jAppender.java (hunk 3 above)
Comment (kamalcph) [reply log4j-appender/src/main/java/org/apache/kafka/log4jappender/KafkaLog4jAppender.java:55]: I was about to create a new Pull request. But, it got merged with this one.

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java (hunk 2 above)
Comment (kamalcph) [reply clients/src/main/java/org/apache/kafka/common/utils/Utils.java:446]: Ok, I'll refactor & remove this method and open a new PR.

File: clients/src/main/java/org/apache/kafka/common/utils/KafkaThread.java (hunk 1 above)
Comment (kamalcph) [reply clients/src/main/java/org/apache/kafka/common/utils/KafkaThread.java:39]: Ok.

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java (hunk 2 above)
Comment (kamalcph) [reply clients/src/main/java/org/apache/kafka/common/utils/Utils.java:446]: In KafkaThread class, Shall I add one overriding constructor `public KafkaThread(final String name)` with daemon as `false` internally ? (In most of the places, the non-daemon thread is being created.)

Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/utils/Utils.java:446]: It's good to make this explicit, so maybe we could have `KafkaThread.daemon` and `KafkaThread.nonDeamon` as static factory methods.

Comment (kamalcph) [reply clients/src/main/java/org/apache/kafka/common/utils/Utils.java:446]: Ok, I'll do that change.
